src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -241,6 +241,36 @@ class PyStubCAS:
     def remove(self, digest: FileDigest | Digest) -> bool: ...
     def action_cache_len(self) -> int: ...
 
+# ------------------------------------------------------------------------------
+# Dependency inference
+# ------------------------------------------------------------------------------
+
+class JavascriptInferenceMetadata:
+    def __init__(self, package_root: str, import_patterns: dict[str, list[str]]) -> None: ...
+    def __eq__(self, other: JavascriptInferenceMetadata | Any) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __repr__(self) -> str: ...
+
+class NativeDependenciesRequest:
+    """A request to parse the dependencies of a file.
+
+    * The `digest` is expected to contain exactly one source file.
+    * Depending on the implementation, a `metadata` structure
+      can be passed. It will be supplied to the native parser, and
+      it will be incorporated into the cache key.
+
+
+    Example:
+        result = await Get(NativeParsedPythonDependencies, NativeDependenciesRequest(input_digest, None)
+    """
+
+    def __init__(
+        self, digest: Digest, metadata: JavascriptInferenceMetadata | None = None
```

Comment:
:grimacing: At this point I'd feel more comfortable with per-language request types. Especially since below this is mirrored on the Rust side.

Reply:
I think I agree. Only downside is that it is more boilerplate and a lot of `Native<language>DependenciesRequest`, probably.